src/firestore/delete.ts

```diff
@@ -269,22 +278,23 @@ export class FirestoreDelete {
     allDescendants: boolean,
     batchSize: number,
     startAfter?: string
-  ): Promise<any[]> {
+  ): Promise<Document[]> {
     const url = this.parent + ":runQuery";
     const body = this.isDocumentPath
       ? this.docDescendantsQuery(allDescendants, batchSize, startAfter)
       : this.collectionDescendantsQuery(allDescendants, batchSize, startAfter);
 
-    return this.apiClient.post<any, Array<{ document?: any }>>(url, body).then((res) => {
+    return this.apiClient.post<any, Array<{ document?: Document }>>(url, body).then((res) => {
       // Return the 'document' property for each element in the response,
       // where it exists.
-      return res.body
-        .filter((x) => {
-          return x.document;
-        })
-        .map((x) => {
-          return x.document;
-        });
+      const docs: Document[] = [];
```

Comment:
This change is because the compiler couldn't infer the the guarantees on the `.document` field from the `.filter`